compiler/src/dotty/tools/dotc/transform/init/Errors.scala

```diff
@@ -40,14 +41,28 @@ object Errors {
             i"$code\t$loc"
           else
             tree.show
+        val positionMarkerLine = positionMarker(pos, indentCount + 3)
 
-        if (last != line)  sb.append(prefix + line + "\n")
+        if (last != line)  sb.append(prefix + line + "\n" + positionMarkerLine )
 
         last = line
       }
       sb.toString
     }
 
+    /** Used to underline source positions in the stack trace
+     */
+    private def positionMarker(pos: SourcePosition, indent: Int): String = {
+      val trimmed = pos.lineContent.takeWhile(c => c.isWhitespace).length
+      val padding = pos.startColumnPadding.substring(trimmed).nn + (" " * indent)
```

Comment:
It looks like this breaks the community build - there are quite a lot of cases when `pos` is `NoPosition` so this will need to be handled somehow